### PR TITLE
Fix PostgreSQL backend race condition when replacing permissions of an object 

### LINKED ISSUE #1210: Race condition in permission.postgresql:replace_object_permissions
Because https://github.com/Kinto/kinto/blob/6ee65b018926d56101015fd1cdf69bf8760d9af8/kinto/core/permission/postgresql/__init__.py#L393-L396 is two queries, it's possible for two queries that both update permissions to race, with the result being that both principals get permissions on the object.

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #1210

### CODE REVIEW

**Comment by leplatrem** [comment]
The failing build on the functional test is very interesting :) We have a edge case that is not covered by unit tests !

File: kinto/core/permission/postgresql/__init__.py [hunk 1]
```diff
@@ -362,38 +362,50 @@ def replace_object_permissions(self, object_id, permissions):
...
 
-        delete_query = """
-        WITH specified_perms AS (
-          VALUES {specified_perms}
-        )
-        DELETE FROM access_control_entries
-         USING specified_perms
-         WHERE object_id = :object_id AND permission = column1
-        """.format(specified_perms=','.join(specified_perms))
+        print(placeholders, specified_perms, new_aces)
```
Comment (glasserc) [inline kinto/core/permission/postgresql/__init__.py]: Leftovers?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -73,7 +73,8 @@ Protocol is now at version **1.16**. See `API changelog`_.
 - Fixed Postgresql errors when specifying empty values in querystring numeric filters. (fixes #1194)
 - Return a 400 Bad Request instead of crashing when the querystring contains bad characters. (fixes #1195)
 - Fix PostgreSQL backend from deleting records of the same name in
-  other buckets and collections when deleting a bucket. (Fixes #1209.)
+  other buckets and collections when deleting a bucket. (fixes #1209)
```
Comment (glasserc) [inline CHANGELOG.rst]: Tangential question: Is this format officially mandated or anything?
